internal/frontend/selfservice.go

```diff
@@ -7,16 +7,33 @@
 package frontend
 
 import (
+	"bytes"
+	"fmt"
+	"image/png"
 	"net/http"
 	"sort"
+	"strconv"
 	"strings"
 
 	"github.com/majewsky/portunus/internal/core"
 	"github.com/majewsky/portunus/internal/crypt"
 	h "github.com/majewsky/portunus/internal/html"
+	"github.com/pquerna/otp"
+	"github.com/pquerna/otp/totp"
 	"github.com/sapcc/go-bits/errext"
 )
 
+const (
+	Memberships    = "memberships"
+	changePassword = "change_password"
+	oldPassword    = "old_password"
+	newPassword    = "new_password"
+	repeatPassword = "repeat_password"
+	sshPublicKeys  = "ssh_public_keys"
+	configureTotp  = "configure_totp"
+	totpToken      = "totp_token"
+)
```

Comment:
I didn't separate this because git commit -p couldn't do it without first undoing the other changes.... I hope this is fine